jcl/src/java.base/share/classes/com/ibm/tools/attach/target/Attachment.java

```diff
@@ -70,17 +70,35 @@
 	private static final class MethodRefsHolder {
 		static Method startLocalManagementAgentMethod = null;
 		static Method startRemoteManagementAgentMethod = null;
+		static final Throwable managementAgentMethodThrowable;
 		static {
-			AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
-				Class<?> agentClass;
-				Class<?> startRemoteArgumentType;
+			managementAgentMethodThrowable = (Throwable) AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
+				String agentClassName = 
+						/*[IF Sidecar19-SE]*/
+						"jdk.internal.agent.Agent"; //$NON-NLS-1$
+						/*[ELSE] Sidecar19-SE
+						"sun.management.Agent"; //$NON-NLS-1$
+						/*[ENDIF] Sidecar19-SE */
+				IPC.logMessage("Loading " + agentClassName); //$NON-NLS-1$
+				Throwable mamtTemp = null;
 				try {
+					Class<?> agentClass = null;
+					Class<?> startRemoteArgumentType = null;
 					/*[IF Sidecar19-SE]*/
-					agentClass = Class.forName("jdk.internal.agent.Agent"); //$NON-NLS-1$
+					String jmaName = "jdk.management.agent"; //$NON-NLS-1$
+					java.lang.Module jmaModule = jdk.internal.module.Modules.loadModule(jmaName);
+					if (null == jmaModule) {
```

Comment:
This should never happen because java.lang.module.FindException will be thrown by loadModule(). Please add a comment.

Reply:
Done.